### PR TITLE
Supports title tag for WordPress 4.4

### PR DESCRIPTION
**Pull request by TweetPressFr**
Since 4.4.0 WP allows theme support for title tag so title_callback won't work. This patch fixes it with new filters.

### CODE REVIEW

File: hm-rewrites.php [hunk 1]
```diff
@@ -269,14 +269,34 @@ public function matched_rule() {
...
-		add_filter( 'wp_title', function( $title, $sep = '' ) use ( $t ) {
+		/**
+		 * Add support for theme support title tag
+		 * @since 4.4.0
+		 */
+		if ( current_theme_supports( 'title-tag' ) ) {
 
-			foreach ( $t->title_callbacks as $callback )
-				$title = call_user_func_array( $callback, array( $title, $sep ) );
+			add_filter( 'pre_get_document_title', function( $title ) use ( $t ) {
```
Comment (rmccue) [inline hm-rewrites.php]: Needs a space after `function`

File: hm-rewrites.php [hunk 2]
```diff
@@ -269,14 +269,34 @@ public function matched_rule() {
...
 
-			return $title;
+				$sep = apply_filters( 'document_title_separator', '-' );
 
-		}, 10, 2 );
+				foreach ( $t->title_callbacks as $callback )
+					$title = call_user_func_array( $callback, array( $title, $sep ) );
```
Comment (rmccue) [inline hm-rewrites.php]: Should add braces while you're here

File: hm-rewrites.php [hunk 3]
```diff
@@ -269,14 +269,34 @@ public function matched_rule() {
...
+
+				return $title;
+
+			} );
+
+		} else {
+			add_filter( 'wp_title', function( $title, $sep = '' ) use ( $t ) {
+
+				foreach ( $t->title_callbacks as $callback )
+					$title = call_user_func_array( $callback, array( $title, $sep ) );
```
Comment (rmccue) [inline hm-rewrites.php]: Braces here too.

File: hm-rewrites.php [hunk 4]
```diff
@@ -269,14 +269,34 @@ public function matched_rule() {
...
-		}, 10, 2 );
+				foreach ( $t->title_callbacks as $callback )
+					$title = call_user_func_array( $callback, array( $title, $sep ) );
+
+				return $title;
+
+			} );
+
+		} else {
+			add_filter( 'wp_title', function( $title, $sep = '' ) use ( $t ) {
```
Comment (rmccue) [inline hm-rewrites.php]: Space after `function`

**Comment by rmccue** [comment]
Hey @TweetPressFr, thanks for the PR! I've left a few inline comments just on some style points that would be great to fix here. :)

**Comment by TweetPressFr** [comment]
Hi Ryan, thanks for the inline comments. Actually the styles are the same as yours if you look at the the other functions.  But I've done what you've asked.

**Comment by rmccue** [comment]
Sorry for not getting back to this earlier, slipped through my inbox. There are a few unrelated commits in here now since it's PR'd from the `master` branch. Can you open a new pull request with the changes separated please?

**Comment by TweetPressFr** [comment]
Yeah, other fixes here :) I'm opening another PR

**Pull request closed without merging**